klient/decoder/decoder.go

```diff
@@ -92,6 +105,30 @@ func DecodeAllFiles(ctx context.Context, fsys fs.FS, pattern string, options ...
 	return objects, err
 }
 
+// ApplyUsingManifestDir resolves all the files in the Directory dirPath against the globbing pattern and creates a kubernetes
+// resource for each of the resources found under the manifest directory.
+func ApplyUsingManifestDir(ctx context.Context, r *resources.Resources, dirPath, pattern string, f PreHandlerFuncHook, createOptions []resources.CreateOption, options ...DecodeOption) error {
+	err := DecodeEachFile(ctx, os.DirFS(dirPath), pattern, func(ctx context.Context, obj k8s.Object) error {
+		if f(ctx, obj) {
```

Comment:
Again, I think this check should not be applied. We assume if someone call this function, it is because they want to apply the files. I don't the additional check is needed.

Reply:
/ack